app/helpers/Intersect.js

```diff
@@ -202,11 +224,15 @@ export function normalizeAndSortLocations(arr) {
     for (let i = 0; i < arr.length; i++) {
       let elem = arr[i];
       if ('time' in elem && 'latitude' in elem && 'longitude' in elem) {
-        result.push({
+        let data = {
           time: Number(elem.time),
           latitude: Number(elem.latitude),
           longitude: Number(elem.longitude),
-        });
+        };
+
+        if ('crossedPaths' in elem) data.crossedPaths = elem.crossedPaths;
```

Comment:
This is scary, copying over crossedPaths which might be mutated elsewhere.
clone the array:

```js
let data = {
  // ..
  crossedPaths: elem.crossedPaths && [...elem.crossedPaths],
}
```